larpandoracontent/LArControlFlow/StitchingCosmicRayMergingTool.cc

```diff
@@ -590,19 +593,138 @@ void StitchingCosmicRayMergingTool::StitchPfos(const MasterAlgorithm *const pAlg
             }
         }
 
-        for (const ParticleFlowObject *const pPfoToDelete : pfoVector)
+        for (const ParticleFlowObject *const pPfoToDelete : reducedPfoVector)
         {
-            if (pPfoToEnlarge == pPfoToDelete)
+            if (selectedPfoToEnlarge == pPfoToDelete)
                 continue;
 
-            pAlgorithm->StitchPfos(pPfoToEnlarge, pPfoToDelete, pfoToLArTPCMap);
-	    stitchedPfosToX0Map.insert(PfoToFloatMap::value_type(pPfoToEnlarge, x0));
+            pAlgorithm->StitchPfos(selectedPfoToEnlarge, pPfoToDelete, pfoToLArTPCMap);
+            stitchedPfosToX0Map.insert(PfoToFloatMap::value_type(selectedPfoToEnlarge, x0));
         }
     }
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+const ParticleFlowObject *StitchingCosmicRayMergingTool::ReduceToLongestStitch(PfoVector &pfoVector, PfoVector &reducedPfoVector,
```

Comment:
(See earlier comment about order of arguments and const qualifier for pfoVector reference.)